.travis.yml

```diff
@@ -287,7 +291,10 @@ jobs:
     env:
       <<: *env-functional-shard_3
       ANSIBLE_VERSION: devel
-
+  - release
+      <<: *if-cron-or-manual-run-or-tagged
+      stage: release
+      script: tox -e upload
```

Comment:
script step should be used for testing, not for publishing. there's deploy step for that.

Reply:
I stopped using deploy when I found how broken and unmaintained were its providers but you are right, there is a deploy script provider that can do the same thing.

I do like more the new stages as it helps organizing much better but I can fix this.

Still, here is something to think about: "tox -e upload" does also run a  twine check which maxes it bit mixed: containing elements of both testing and uploading / deploying.

Because you are not running all tests on each commit I wonder if it would not be better to migrate to a multe-branch setup when PRs are made on `devel` branch and once a day we run the heavy tests and when they pass, we merge the changes to master. 

Something like this could assure that master always contains code that passed the heavy-testing, while keeping the roundtrip for testing and merging a PRs short.

Different users may pick their own poison.

Reply:
@ssbarnea in my implementation, check is taken care of and it runs from the proper place. And yes, I use stages.

I have a huge experience with Travis CI and I know about drawbacks as well but I also think that abusing tox this is not the answer since it limits our ability to easily distinguish between different actions happening in logs, for example.

Multi-branch sounds interesting (like the original Git flow?) but this is not the place to discuss it. Feel free to create a separate issue about it, currently cron takes care of it + the way I set up conditions facilitates the flow.

Please let me take care of CI+packaging because it's what I'm good at.

OTOH I could use some help with enabling more tests in CI, I'm struggling with #1591 so if you want to help this would be ideal if you could help with that :)